pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -113,13 +115,40 @@ func TestGetNodeCondition(t *testing.T) {
 	}
 }
 
+func assertEvents(t *testing.T, testCase string, expectedEvents []string, realEvents chan string) {
+	if len(expectedEvents) != len(realEvents) {
+		t.Errorf(
+			"Test case: %s. Number of expected events (%v) differs from number of real events (%v)",
+			testCase,
+			len(expectedEvents),
+			len(realEvents),
+		)
+	} else {
+		for _, eventType := range expectedEvents {
+			select {
+			case event := <-realEvents:
+				if !strings.Contains(event, fmt.Sprintf(" %s ", eventType)) {
+					t.Errorf("Test case: %s. Expected %v event, got: %v", testCase, eventType, event)
+				}
+			default:
+				t.Errorf("Test case: %s. Expected %v event, but no event occured", testCase, eventType)
+			}
+		}
+	}
+}
+
 // newFakeReconciler returns a new reconcile.Reconciler with a fake client
 func newFakeReconciler(initObjects ...runtime.Object) *ReconcileMachineHealthCheck {
+	return newFakeReconcilerCustomRecorder(nil, initObjects...)
+}
+
+func newFakeReconcilerCustomRecorder(recorder record.EventRecorder, initObjects ...runtime.Object) *ReconcileMachineHealthCheck {
 	fakeClient := fake.NewFakeClient(initObjects...)
```

Comment:
why is this called newFakeReconcilerCustomRecorder when it returns a `ReconcileMachineHealthCheck`?
we can probably have only one `func newFakeReconciler ` which takes event number as a parameter an instantiate the recorded if it gets a value

Reply:
I originally done it the same way as you are proposing. But then in tests you can get a recorder instance only from reconciler. This will not work as `reconciler.recorder` will be of type `EventRecorder` which doesn't have way to get events from it (unlike `FakeRecorder`).
Name of the function was misleading. I should be `newFakeReconcilerWithCustomRecorder`.